src/workshed/index.ts

```diff
@@ -0,0 +1,165 @@
+}
+
+function currentWorkshed(): GarboWorkshed {
+  return getGarboWorkshed(getWorkshed());
+}
+
+let _nextWorkshed: GarboWorkshed | null = null;
+function nextWorkshed(): GarboWorkshed {
+  _nextWorkshed ??= getGarboWorkshed(stringToWorkshedItem());
+  return _nextWorkshed ?? defaultWorkshed;
+}
+
+const defaultWorkshed = new GarboWorkshed($item`none`);
+
+const worksheds = [
+  new GarboWorkshed(
+    $item`model train set`,
+    () => {
+      // We should always get value from the trainset, so we would never switch from it
+      return false;
+    },
+    () => {
+      if (!isTrainsetConfigurable()) return;
+      if (get("trainsetConfiguration") === "") {
+        print("Reconfiguring trainset, as our next station is empty", "blue");
+        return setTrainsetConfiguration(defaultPieces);
+      } else {
+        const pieces = getTrainsetConfiguration();
+        const offset = toInt(get("trainsetPosition")) % 8;
+        const nextPiece = pieces[offset];
+
+        if ([TrainsetPiece.DOUBLE_NEXT_STATION, TrainsetPiece.GAIN_MEAT].includes(nextPiece)) {
+          return;
+        }
+        print(`Reconfiguring trainset, as our next station is ${String(nextPiece)}`, "blue");
+        return setTrainsetConfiguration(offsetDefaultPieces(offset));
+      }
+    }
+  ),
+  new GarboWorkshed(
+    $item`cold medicine cabinet`,
+    () => {
+      return get("_coldMedicineConsults") >= 5;
+    },
+    () => {
+      if (get("_nextColdMedicineConsult") > totalTurnsPlayed()) return;
+      grabMedicine();
+    }
+  ),
+  new GarboWorkshed(
+    $item`Asdon Martin keyfob`,
+    () => {
+      return (
+        haveEffect($effect`Driving Observantly`) >=
+        estimatedTurns() +
+          (globalOptions.ascend ? 0 : 400 + clamp((get("valueOfAdventure") - 4000) / 8, 0, 600))
+      );
+    },
+    () => {
+      AsdonMartin.drive(
+        $effect`Driving Observantly`,
+        estimatedTurns() +
+          (globalOptions.ascend ? 0 : 400 + clamp((get("valueOfAdventure") - 4000) / 8, 0, 600))
+      );
+    }
+  ),
+  new GarboWorkshed(
+    $item`Little Geneticist DNA-Splicing Lab`,
+    () => {
+      // This will likely always return true or false for now, depending on the start state of garbo
+      // Since we don't actually support using the syringe in combat at this time, the counter will never change
+      return get("_dnaPotionsMade") >= 3;
+    },
+    () => {
+      // Just grab whatever tonics for now, since we don't actually have support for DNA
+      if (get("dnaSyringe")) DNALab.makeTonic(3);
+    }
+  ),
+  new GarboWorkshed(
+    $item`spinning wheel`,
+    () => {
+      return get("_spinningWheel");
+    },
+    () => {
+      // We simply assume you will not gain a level while garboing, since we do not do powerlevellings
+      // So we will just use the spinning wheel immediately
+      visitUrl("campground.php?action=spinningwheel");
+    }
```

Comment:
If this isn't in daily, put it in daily

Reply:
I don't really see a need though - we would still need this here if low-end users are swapping into the spinning wheel